Sources/Group.php

```diff
@@ -1077,15 +1081,16 @@ public function addMembers(int|array $members, string $type = 'auto', bool $perm
 
 			$to_set = [];
 
+			/** @var \SMF\User $member */
 			foreach ($set_primary as $member) {
-				$new_additional_groups = array_diff(User::$loaded[$member]->groups, [$this->id, User::$loaded[$member]->post_group_id]);
+				$new_additional_groups = array_diff(User::$loaded[$member->id]->groups, [$this->id, User::$loaded[$member->id]->post_group_id]);
 				sort($new_additional_groups);
 
 				$to_set[implode(',', $new_additional_groups)][] = $member->id;
 			}
 
-			foreach ($to_set as $new_additional_groups => $member_ids) {
-				User::updateMemberData($member_ids, [
+			foreach ($to_set as $new_additional_groups => $mems) {
```

Comment:
Not to be picky here but can we get a more descriptive $var here, such as $mem_ids and $mem_id. or $ids and $id. Anything to communicate what its actually using.

Reply:
I will put it back to `$member_ids`.  It is an array of ints containing the `id_member` to be processed.